service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateStorageAccountCorsRules.java

```diff
@@ -0,0 +1,118 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.library.landingzones.definition.factories.ParametersResolver;
+import bio.terra.landingzone.library.landingzones.definition.factories.parameters.ParametersExtractor;
+import bio.terra.landingzone.library.landingzones.definition.factories.parameters.StorageAccountBlobCorsParametersNames;
+import bio.terra.landingzone.stairway.flight.LandingZoneFlightMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import com.azure.resourcemanager.storage.models.CorsRule;
+import com.azure.resourcemanager.storage.models.CorsRuleAllowedMethodsItem;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateStorageAccountCorsRules extends BaseResourceCreateStep {
+  private static final Logger logger = LoggerFactory.getLogger(CreateStorageAccountCorsRules.class);
+
+  public CreateStorageAccountCorsRules(
+      ArmManagers armManagers,
+      ParametersResolver parametersResolver,
+      ResourceNameGenerator resourceNameGenerator) {
+    super(armManagers, parametersResolver, resourceNameGenerator);
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    // will be removed as part of storage account deletion
```

Comment:
I wonder if this step should just be collapsed into the StorageAccount creation step. Do we anticipate ever wanting to do one without the other?

Reply:
Current API doesn't allow to attach CORS rules together with creation. The idea (if I am not mistake in WSM we do it the same way) is to have separate step for each resource/option. But we might consider changing it.